src/main/scala/TargetProfile.scala

```diff
@@ -0,0 +1,19 @@
+// Copyright (c) 2019 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package basic
+
+import basic.misc._
+import gem.enum._
+
+/** Target properties we need to know at phase zero. */
+final case class TargetProfile(
+  spatialProfile:       SpatialProfile,
+  spectralDistribution: SpectralDistribution,
+  magnitude:            Double,
```

Comment:
I've never understood magnitudes and systems very well. I wonder if there is a way to represent magnitude as an integer?  I suppose it depends on the system.  I wonder if there should be three kinds of magnitude with their own brightness value instead of a `magnitude` `Double` and a `MagnitudeSystem` `enum`.

Reply:
There are actually six magnitude systems, plus an additional six with the units squared for surface brightness. We only have three in the OT but ITC supports all six and from talking with Andy I think we need to as well.

A problem here is that the units are all extremely weird and I haven't found a way to represent them. They can probably all be scaled `Long`s but I'd like to return to this as a separate task.

Reply:
We have a `MagnitudeValue` class in OCS3 but it's half-baked. I like the ADT idea.